### PR TITLE
Add thomas's PHCCP user and code deploy group

### PR DESCRIPTION
**Pull request by tschaffter**
Thanks!

### CODE REVIEW

**Comment by zaro0508** [comment]
Could you please provide a description on the intent of this change?  is the intention to create a service user to allow travis to deploy stuff to sandbox ?

**Comment by tschaffter** [comment]
@zaro0508 We are putting in place an auto-deployment pipeline for the stack that we are developing for Roche. The architecture of this pipeline is described in this diagram: https://www.lucidchart.com/documents/view/fe124033-996c-4a72-ba52-abba6504f44e/0

The following piece of code will be inserted at the bottom of my Travis file:

```
deploy:
  - provider: script
    script: docker push "${IMAGE_NAME}:develop" && docker push "${IMAGE_NAME}:${git_sha}-develop"
    on:
      branch: develop
  - provider: codedeploy
    revision_type: github
    access_key_id: "$ACCESS_KEY_ID"
    secret_access_key: "$SECRET_ACCESS_KEY"
    application: "phccp-autodeploy"
    deployment_group: "phccp-autodeploy-group"
    region: "us-east-1"
    on:
      branch: develop
```

The second provider object will be used to trigger AWS CodeDeploy to notify it that our stack should be restarted to run with the latest version of the docker image, which has been pushed (see above Travis snippet). In order to trigger CodeDeploy, the Travis file need the access key id and secret access key for the AWS User that I want to create with this PR. Also, this PR creates the group `AWSIAMCodeDeployUsersGroup` that provide the policy `arn:aws:iam::aws:policy/AWSCodeDeployFullAccess`. The new user is then added to this group to have sufficient access to trigger CodeDeploy.

File: templates/accounts.yaml [hunk 1]
```diff
@@ -172,6 +172,15 @@ Resources:
       LoginProfile:
         Password: !Ref InitNewUserPassword
         PasswordResetRequired: true
+  AWSIAMThomasSchaffterPhccpAutoDeployUser:
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply templates/accounts.yaml]: `PhccpAutoDeployUser` seems safer to me than reusing an `AppDeployerUser`. I don't want anybody else to be able to interfere with the deployment pipeline setup for this project.

File: templates/accounts.yaml [hunk 2]
```diff
@@ -172,6 +172,15 @@ Resources:
...
+    Type: 'AWS::IAM::User'
+    Properties:
+      UserName: thomas.schaffter+phccp-autodeploy@sagebase.org
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply templates/accounts.yaml]: @zaro0508 A username is defined for all the other Users. Why not this one? I'll however address your request.

File: templates/accounts.yaml [hunk 3]
```diff
@@ -172,6 +172,15 @@ Resources:
...
+      Groups:
+        - !Ref AWSIAMCodeDeployUsersGroup
+      LoginProfile:
```
[the remark this answers is not in the document]
Comment (tschaffter) [reply templates/accounts.yaml]: @zaro0508 I can remove `LoginProfile`. But how will I receive the client key and secret key?

**Comment by tschaffter** [comment]
@zaro0508 

> 1. The user you created will need to have an access key. do you plan to create that here, as part of this automation, or do you plan to do that manually?

I was planning to do it using AWS console. But if I remove `LoginProfile` as requested, can you clarify how I will obtain the key?

> 2. Directions for codedeploy's deployment_group[1] requires setting up a role[2]. I don't see that been done in this change. Do you plan to add that or has that been created manually?

I don't know about that. I'm following the instructions from a Roche engineer who successfully deployed the pipeline on his end for testing purpose. Creating a role with CodeDeploy access and then adding a user to this role makes sense to me, but there is maybe another way to do it. Unless this is critical, I would suggest to go with the current strategy.

**Comment by tschaffter** [comment]
@zaro0508 I have updated my branch to address all your requests. Do I need to do anything else regarding this PR (I don't have much experience with PR yet)? Thanks!

File: templates/accounts.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tschaffter) [reply templates/accounts.yaml]: Renamed the user to `PhccpAutoDeployUser`.

File: templates/accounts.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (tschaffter) [reply templates/accounts.yaml]: Removed `UserName`

File: templates/accounts.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (tschaffter) [reply templates/accounts.yaml]: Removed `LoginProfile`

**Comment by zaro0508** [comment]
@tschaffter  I will help with this PR.

**Comment by tschaffter** [comment]
@zaro0508 Is there anything that I still need to address?

File: templates/accounts.yaml [hunk 4]
```diff
@@ -437,3 +449,15 @@ Outputs:
     Value: !GetAtt AWSIAMMssmUserAccessKey.SecretAccessKey
     Export:
       Name: !Sub '${AWS::Region}-${AWS::StackName}-MssmUserSecretAccessKey'
+  PhccpServiceUserAccessKey:
```
Comment (tschaffter) [inline templates/accounts.yaml]: @zaro0508 Will running this generate an access key and secret? Will I receive the key and secret by email? Thx!

**Comment by tschaffter** [comment]
@zaro0508 I'm fine with the changes.

File: templates/accounts.yaml (hunk 4 above)
Comment (zaro0508) [reply templates/accounts.yaml]: yes and yes

**Pull request merged**